backend/matching-service/src/matchQueue.js

```diff
@@ -0,0 +1,75 @@
+import { UserMatchRequest } from "./models/UserMatchRequest.js";
+import { getChannel } from "./connections.js";
+
+let channel;
+
+export const initializeMatchQueue = async () => {
+  channel = getChannel();
+  if (!channel) {
+    throw new Error("No Rabbitmq channel initialized");
+  }
+};
+
+export const removeUser = async (userId) => {
+  await UserMatchRequest.deleteOne({ userId });
+  const message = `{"userId":"${userId}","action":"remove"}`;
+  channel.sendToQueue("match_requests", Buffer.from(message), {
+    persistent: true,
+  });
+};
+
+// Add user to the queue and match it
+export const addUser = async (user) => {
+  try {
+    const checkExisting = await UserMatchRequest.findOne({
+      userId: user.userId,
+    });
+    if (checkExisting) {
+      console.log(`User ${user.userId} already has an active session.`);
+      updateUser(user.userId, "error: You can only request one session at a time.");
+      return { error: "You can only request one session at a time." };
+    }
+
+    const userMatchRequest = new UserMatchRequest(user);
+    console.log("User added:", userMatchRequest);
+    await userMatchRequest.save();
+    channel.sendToQueue(
+      "match_requests",
+      Buffer.from(JSON.stringify(userMatchRequest)),
+      { persistent: true }
+    );
+
+    setTimeout(async () => {
+      const existingUser = await UserMatchRequest.findOne({
+        userId: userMatchRequest.userId,
+      });
+      if (existingUser) {
+        updateUser(userMatchRequest.userId, "timeout");
+        await UserMatchRequest.deleteOne({ userId: userMatchRequest.userId });
+      }
+    }, 60000);
```

Comment:
For the Timeout duration, 60 seconds may be too long. The Project Documentation and our plan in D1 are both 30 seconds. You may change the setTimeout duration from `60000` to `30000`.